From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Doe <john.doe@somewhere.on.planet>
Date: Sun, 14 Jan 2024 11:44:47 -0800
Subject: Patching kernel sunxi64 files
 arch/arm64/boot/dts/allwinner/overlay/Makefile
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-fixup.scr-cmd
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-i2c3.dtso
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-ir.dtso
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-otg-host.dtso
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-spi1-cs0-spidev.dtso
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-uart5.dtso

Signed-off-by: John Doe <john.doe@somewhere.on.planet>
---
 arch/arm64/boot/dts/allwinner/overlay/Makefile                                    | 11 +++--
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-fixup.scr-cmd        | 15 +++++++
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-i2c3.dtso            |  8 ++++
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-ir.dtso              |  8 ++++
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-otg-host.dtso        | 19 +++++++++
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-spi1-cs0-spidev.dtso | 21 ++++++++++
 arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-uart5.dtso           | 16 +++++++
 7 files changed, 95 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/overlay/Makefile b/arch/arm64/boot/dts/allwinner/overlay/Makefile
index 84711585f..7d866ae56 100644
--- a/arch/arm64/boot/dts/allwinner/overlay/Makefile
+++ b/arch/arm64/boot/dts/allwinner/overlay/Makefile
@@ -56,17 +56,22 @@ dtb-$(CONFIG_ARCH_SUNXI) += \
 	sun50i-h616-spidev1_2.dtbo \
 	sun50i-h616-ir.dtbo \
 	sun50i-h616-tft35_spi.dtbo \
 	sun50i-h616-mcp2515.dtbo \
 	sun50i-h616-ws2812.dtbo \
-	sun50i-h616-light.dtbo
-
+	sun50i-h616-light.dtbo \
+	sun50i-orangepizero2-3-i2c3.dtbo \
+	sun50i-orangepizero2-3-ir.dtbo \
+	sun50i-orangepizero2-3-otg-host.dtbo \
+	sun50i-orangepizero2-3-spi1-cs0-spidev.dtbo \
+	sun50i-orangepizero2-3-uart5.dtbo
 scr-$(CONFIG_ARCH_SUNXI) += \
 	sun50i-a64-fixup.scr \
 	sun50i-h5-fixup.scr \
 	sun50i-h6-fixup.scr \
-	sun50i-h616-fixup.scr
+	sun50i-h616-fixup.scr \
+	sun50i-orangepizero2-3-fixup.scr
 
 dtbotxt-$(CONFIG_ARCH_SUNXI) += \
 	README.sun50i-a64-overlays \
 	README.sun50i-h5-overlays
 
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-fixup.scr-cmd b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-fixup.scr-cmd
new file mode 100755
index 000000000..58ddc990a
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-fixup.scr-cmd
@@ -0,0 +1,15 @@
+# overlays fixup script
+# implements (or rather substitutes) overlay arguments functionality
+# using u-boot scripting, environment variables and "fdt" command
+
+# setexpr test_var ${tmp_bank} - A
+# works only for hex numbers (A-F)
+
+setenv decompose_pin 'setexpr tmp_bank sub "P(C|G|H|I)\\d+" "\\1";
+setexpr tmp_pin sub "P\\S(\\d+)" "\\1";
+test "${tmp_bank}" = "C" && setenv tmp_bank 2;
+test "${tmp_bank}" = "G" && setenv tmp_bank 6'
+test "${tmp_bank}" = "H" && setenv tmp_bank 7;
+test "${tmp_bank}" = "I" && setenv tmp_bank 8;
+
+# no fixups required
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-i2c3.dtso b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-i2c3.dtso
new file mode 100644
index 000000000..4438f08d1
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-i2c3.dtso
@@ -0,0 +1,8 @@
+/dts-v1/;
+/plugin/;
+
+&i2c3 {
+        status = "okay";
+        pinctrl-names = "default";
+        pinctrl-0 = <&i2c3_ph_pins>;
+}; 
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-ir.dtso b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-ir.dtso
new file mode 100644
index 000000000..75ae7c6a2
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-ir.dtso
@@ -0,0 +1,8 @@
+/dts-v1/;
+/plugin/;
+
+&ir {
+        status = "okay";
+        pinctrl-names = "default";
+        pinctrl-0 = <&ir_rx_pin>;
+}; 
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-otg-host.dtso b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-otg-host.dtso
new file mode 100644
index 000000000..c0fe0f3db
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-otg-host.dtso
@@ -0,0 +1,19 @@
+/dts-v1/;
+/plugin/;
+
+&usbotg {
+		dr_mode = "host";
+                status = "okay";
+}; 
+
+&usbphy {
+                status = "okay";
+};
+
+&ehci0 {
+                status = "okay";
+};
+
+&ohci0 {
+                status = "okay";
+};
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-spi1-cs0-spidev.dtso b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-spi1-cs0-spidev.dtso
new file mode 100644
index 000000000..2b4995204
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-spi1-cs0-spidev.dtso
@@ -0,0 +1,21 @@
+/dts-v1/;
+/plugin/;
+
+&spi1_cs0_pin {
+	pins = "PH9";
+};
+
+&spi1 {
+	status = "okay";
+        #address-cells = <0x01>;
+        #size-cells = <0x00>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&spi1_pins &spi1_cs0_pin>;
+
+	       spidev@0 {
+	               compatible = "rohm,dh2228fv";
+                       status = "okay";
+                       reg = <0x00>;
+                       spi-max-frequency = <0xf4240>;
+               };
+}; 
diff --git a/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-uart5.dtso b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-uart5.dtso
new file mode 100644
index 000000000..985023089
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/overlay/sun50i-orangepizero2-3-uart5.dtso
@@ -0,0 +1,16 @@
+/dts-v1/;
+/plugin/;
+
+&{/soc/pinctrl@300b000} {
+
+	uart5_ph_pins: uart5-ph-pins {
+		pins = "PH2\0PH3";
+		function = "uart5";
+	};
+};
+
+&uart5 {
+        status = "okay";
+        pinctrl-names = "default";
+        pinctrl-0 = <&uart5_ph_pins>;
+}; 
-- 
Created with Armbian build tools https://github.com/armbian/build
